dali/imgcodec/decoders/tiff_libtiff.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_IMGCODEC_DECODERS_TIFF_LIBTIFF_H_
+#define DALI_IMGCODEC_DECODERS_TIFF_LIBTIFF_H_
+
+#include <memory>
+#include "dali/imgcodec/image_decoder.h"
+#include "dali/imgcodec/decoders/decoder_parallel_impl.h"
+
+namespace dali {
+namespace imgcodec {
+
+class DLL_PUBLIC LibTiffDecoderInstance : public BatchParallelDecoderImpl {
+ public:
+  using Base = BatchParallelDecoderImpl;
+  LibTiffDecoderInstance(int device_id, ThreadPool *tp) : Base(device_id, tp) {}
+
+  using Base::Decode;
+  DecodeResult Decode(SampleView<CPUBackend> out, ImageSource *in,
+                      DecodeParams opts, const ROI &roi) override;
+};
+
+class LibTiffDecoder : public ImageDecoder {
+ public:
+  ImageDecoderProperties GetProperties() const override {
+    static const auto props = []() {
+      ImageDecoderProperties props;
+      props.supported_input_kinds = InputKind::DeviceMemory | InputKind::HostMemory
```

Comment:
```suggestion
      props.supported_input_kinds = InputKind::HostMemory
```
I don't think we can support DeviceMemory in this implementation

Reply:
You're right, thanks. Btw this PR is really in progress (its more of a code dump than a PR, just to have some example for https://github.com/NVIDIA/DALI/pull/4103), so I think reviewing it is not worth your time yet ;)